docs_src/design/adr/system-management/0006-Metrics-Collection.md

```diff
@@ -0,0 +1,382 @@
+
+#### Security
+
+Security metrics may be more difficult to ascertain as they are cross service metrics.  They may have to be dealt with per service.  Also, true threat detection based on metrics may be a feature best provided by 3rd party based on particular threats and security profile needs.
+
+- Number of API requests denied due to wrong access token (Kong) per service and within a given time
+- Number of secrets accessed per service name
+- Count of any accesses and failures to the data persistence layer
+- Count of service start and restart attempts
+
+### Design Proposal
+
+#### Collect and Push Architecture
+
+Metric data will be collected and cached by each service.  At designated times (kicked off by configurable schedule), the service will collect telemetry data from the cache and push it to a designated message bus topic.
+
+#### Metrics Messaging
+
+Cached metric data, at the designated time, will be marshaled into a message and pushed to the pre-configured message bus topic.
+
+Each metric message consists of several key/value pairs:
+- a **required** name (the name of the metric) such as service-uptime
+- a **required** value which is the telemetry value collected such as 120 as the number of hours the service has been up.
+- a **required** timestamp is the time (in Epoch timestamp/milliseconds format) at which the data was collected (similar in nature to the origin of sensed data). 
+- an optional collection (array) of tags.  The tags are sets of key/value pairs of strings that provide amplifying information about the telemetry.  Tags may include:
+    - originating service name
+    - unit of measure associated with the telemetry value
+    - value type of the value
+    - additional values when the metric is more than just one value (example: when using a histogram, it would include min, max, mean and sum values)
+
+The metric name must be unique for that service.  Because some metrics are reported from multiple services (such as service uptime), the name is not required to be unique across all services.
+
+All information (keys, values, tags, etc.) is in string format and placed in a JSON array within the message body.  Here are some example representations:
+
+**Example metric message body with a single value**
+{"name":"service-up", "value":"120", "timestamp":"1602168089665570000", "tags":{"service":"coredata","uom":"days","type":"int64"}}
+
+**Example metric message body with multiple values**
+{"name":"api-requests", "value":"24", "timestamp":"1602168089665570001", "tags":{"service":"coredata","uom":"count","type":"int64", "mean":"0.0665", "rate1":"0.111", "rate5":"0.150","rate15":"0.111"}}
+
+!!! Note
+    The key or metric name must be unique when using go-metrics as it requires the metric name to be unique per the registry.  Metrics are considered immutable.

```

Comment:
unique across an EdgeX instance or individual EdgeX service?

Reply:
@tonyespy , unique to the individual service.